cli/src/main.rs

```diff
@@ -4,13 +4,12 @@ use std::process::Command;
 
 use base64;
 use clap::{crate_name, crate_version, App, AppSettings, SubCommand};
-use log::error;
 use serde_json::json;
 
-use anycloud::deploy::{get_config, info, new, terminate, upgrade, ALAN_VERSION};
+use anycloud::deploy::{client_error, get_config, info, new, terminate, upgrade, ALAN_VERSION};
 use anycloud::oauth::get_token;
 
-fn get_dockerfile_b64() -> String {
+async fn get_dockerfile_b64(token: &str, cluster_id: Option<&str>) -> String {
```

Comment:
Why don't we make a `OnceCell` with the `token` so we don't have to pass it to every function? I think it's fair to consider it part of the global state of the program...

Reply:
Since this is a CLI, the token outlives the program and we need to re-use it over and over again so we store it in a file https://github.com/alantech/anycloud/blob/main/cli/src/oauth/mod.rs. If there is a better way than that, we should do it. I couldn't think of one though

Reply:
No, he's saying that for the duration of the program once that token is loaded into memory it is static for the lifetime of the program, so it could be a `OnceCell` global instead.

Reply:
I can't either. This is just for accessing the `token` during the lifetime of the CLI process. It's easier to have the static value instead of passing around `token` to every single function or otherwise cause large diffs when functions don't need `token` until they call a function that does.

Reply:
Yes, agree it is not the cleanest way. Will take a look to the `OneCell` option once the `token` is loaded.

Reply:
ah got it! 💯